Misc ui improvements

tcourdy opened the pull request:
Making this a PR now so it can be merged in quicker.  All of the items are done except two, which I will look into once I'm done looking at Sam's PR.

1.) Enabling the stop button before user presses start causes undefined error if user actually clicks the stop button.

2.) Toolbar doesn't refresh properly on detach.

Closes #122 
Closes #129

tcourdy committed:
Miscellaneous UI Polish
--Back button to functions on debugger view
--Double clicking an executable launches it
--Change "Functions" to "Breakpoints"
--Disable call button until all parameters are set
--Change "Debug Trace List" to "Trace History"
--Vertical center line numbers for source code
--Auto focus to parameter input field to make it clear the user has to
enter something

tcourdy committed:
Trace History UI polish
--Fix trace history counter
--Fix undefined error when program ends and angular tries to redraw the
trace history component

tcourdy committed:
Fix bug that always says not attached in toolbar

tcourdy committed:
More UI polish
-Fix wacky source code line widths when expandingn loop
-Enter key will call functions when all parameters are filled out
-Call button disabled until all parameters are entered

tcourdy committed:
'Pause' button says 'Cancel' on function execution

tcourdy committed:
Fixed stop button bug

tcourdy committed:
Fix prismjs and dark theme

tcourdy committed:
Fixed source code spacing issue on debugger component

elliothatch reviewed (approved):
Skimmed it, looks okay

tcourdy committed:
Update flex-layout package version to fix source code viewer bug

rpjohnst committed:
UI cleanup

- clear DebuggerService's currentExecution on cancel and return
  (disables pause button after trace completion)
- base start/continue button on DebuggerComponent currentExecution
  (fixes the button after the above change)
- check for an execution before attempting to stop one in KillProcess
- check that sourceTypes actually contains parameter type
  (fixes a crash in trace history viewer after program exit)
